src/main/java/com/minecolonies/api/tools/registry/ToolTypeEntry.java

```diff
@@ -0,0 +1,216 @@
+package com.minecolonies.api.tools.registry;
+
+import com.minecolonies.api.MinecoloniesAPIProxy;
+import com.minecolonies.api.tools.ModToolTypes;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * An entry in the ToolType registry that defines the types of
+ * tools within the colony.
+ */
+public final class ToolTypeEntry
+{
+    /**
+     * The registry identifier for this tool type.
+     */
+    private final ResourceLocation registryName;
+
+    /**
+     * The name of the tool type.
+     */
+    private final String name;
```

Comment:
Name is obsolete and should not be used at all, that's what we have the registry name for.

Reply:
Done.